[DOCS] Update SQL REST API pages for new structure

jrodewig opened the pull request:
#43007 restructured the SQL REST API docs so they display across several pages.

This updates up a reference that assumes a single page in the "Paginating through a large response" section. It also reformats a tip for the Kibana console.

Closes #50688

### Preview of changes
http://elasticsearch_50690.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/sql-rest.html

jrodewig committed:
[DOCS] Reformat SQL REST API page

Re-adds several `[float]` attributes removed in #43007. Also adds
several `[discrete]` attributes for new subsections.

These attributes ensure the sections in this file display on the
same HTML page of the docs output.

elasticmachine commented:
Pinging @elastic/es-search (:Search/SQL)

elasticmachine commented:
Pinging @elastic/es-docs (>docs)

astefan reviewed (changes requested):
Thank you for catching that change that made the REST page look a bit off, at that time I didn't spot it. I wrongly assumed it's the overall docs layout change that introduced it.

These being said, though, I would prefer keeping the current layout, but maybe changing the first sentence from the problematic page. Instead of
> Using the example above, one can continue to the next page by sending back the cursor field.

how about using

> Using the example from the previous section, one can continue to the next page by sending back the cursor field.

astefan commented:
One more nitpick, since it's in the same area, feel free to ignore if you believe it should belong in a different PR and I'll open one:

![Capture](https://user-images.githubusercontent.com/893749/71888191-439d2580-3148-11ea-80a5-26db5b38cb8b.PNG)

I don't remember how "Using Kibana console" small bit of info looked like when it was introduced, but now it doesn't look right. If it's not too much of a change, can you make it a `NOTE` maybe? (ie not a sub-section per se and still belonging to "Overview" sub-section, but distinct enough that it stands out a bit from the rest of the text in "Overview").

jrodewig committed:
Revert "[DOCS] Reformat SQL REST API page"

This reverts commit 06e732854adc5814f102fb52071bfd1f4d91114b.

jrodewig committed:
Address review feedback

jrodewig commented:
Thanks @astefan 

With 7fb17c670791b85a27ee8e19076a5b889c1ac399, I've updated the first sentence of the "Paginating..." section. That commit also formats the Kibana info as a tip:

<img width="748" alt="Screen Shot 2020-01-07 at 7 44 17 AM" src="https://user-images.githubusercontent.com/40268737/71899707-e2625b80-3121-11ea-8eff-6512632b756e.png">

I also removed all the previously added `[float]` and `[discrete]` attributes so the docs structure will remain the same.

astefan commented on docs/reference/sql/endpoints/rest.asciidoc:
```diff
@@ -270,8 +273,8 @@ cursor: "sDXF1ZXJ5QW5kRmV0Y2gBAAAAAAAAAAEWWWdrRlVfSS1TbDYtcW9lc1FJNmlYdw==:BAFmB
-Using the example above, one can continue to the next page by sending back the `cursor` field. In
-case of text format the cursor is returned as `Cursor` http header.
+Using the example from the <<sql-rest-format,previous section>>, one can
+continue to the next page by sending back the cursor field.
```
Please, also, keep the second sentence as well. The final text should be

> Using the example from the <<sql-rest-format,previous section>>, one can continue to the next page by sending back the `cursor` field. In case of text format the cursor is returned as `Cursor` http header.

jrodewig replied:
Thanks for catching!

astefan reviewed (approved):
Left one more comment. Otherwise LGTM. Many thanks for addressing these.

jrodewig committed:
fix overactive copy/paste

jrodewig commented:
### Backport commits
master: a5b6242
7.x: 0753915eedde39c791967a65189aa2d95f06409f
7.5: 80e4022b44c7bad27584fbd0ba4eab55835eacd6
6.8: 338a5381f57
